#
# 30014567 - window change events not passed when using session multiplexing
# Upstreadm code was dropping too many basic privileges.
#
# Logged as upstream bug https://bugzilla.mindrot.org/show_bug.cgi?id=3030
# Pull request: https://github.com/openssh/openssh-portable/pull/135
#
diff -ur old/openbsd-compat/port-solaris.c new/openbsd-compat/port-solaris.c
--- old/openbsd-compat/port-solaris.c	2019-07-05 12:21:15.262497877 +0000
+++ new/openbsd-compat/port-solaris.c	2019-07-05 13:48:19.145472027 +0000
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 2006 Chad Mynhier.
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
  *
  * Permission to use, copy, modify, and distribute this software for any
  * purpose with or without fee is hereby granted, provided that the above
@@ -398,8 +399,7 @@
 	    priv_delset(pset, PRIV_NET_ACCESS) != 0 ||
 #endif
 	    priv_delset(pset, PRIV_PROC_EXEC) != 0 ||
-	    priv_delset(pset, PRIV_PROC_INFO) != 0 ||
-	    priv_delset(pset, PRIV_PROC_SESSION) != 0)
+	    priv_delset(pset, PRIV_PROC_INFO) != 0)
 		fatal("priv_delset: %s", strerror(errno));
 
 	if (setppriv(PRIV_SET, PRIV_PERMITTED, pset) != 0 ||
